gui/wxpython/gui_core/prompt.py

```diff
@@ -301,10 +313,20 @@ def OnItemSelected(self, event):
                 self.cmdDesc = None
 
     def OnKillFocus(self, event):
-        """Hides autocomplete"""
+        """Hides autocomplete and writes a hint if a control is currently empty."""
         # hide autocomplete
         if self.AutoCompActive():
             self.AutoCompCancel()
+        if self.IsEmpty():
+            self.StyleSetForeground(0, self.text_color["hint"])
+            self.WriteText(self.hint)
```

Comment:
These could be methods ShowHint and HideHint so you don't need to repeat it upon initialization and the colors can be defined in those function so you won't need self.text_color.